doc/guides/usage-of-primordials-in-core.md

```diff
@@ -0,0 +1,256 @@
+# Usage of primordials in core
+
+The file `lib/internal/per_context/primordials.js` subclasses and stores the JS
+builtins that come from the VM so that Node.js builtin modules do not need to
+later look these up from the global proxy, which can be mutated by users.
+
+Usage of primordials should be preferred for any new code, but replacing current
+code with primordials should be done with care.
+
+## Accessing primoridals
+
+The primordials are meant for internal use only, and are only accessible for
+internal core modules. User-code cannot use or rely on primordials. It is usually
+fine to rely on ECMAScript built-ins and assume that it will behave as specified.
+
+If you'd like to access the `primordials` object to help you with Node.js core
+development or for tinkering, you can make it globally available:
+
+```bash
+node --expose-internals -r internal/test/binding
+```
+
+## Contents of primoridals
+
+### Properties of the global object
+
+Objects and functions on global objects can be deleted or replaced. Using them
+from primordials makes the code more reliable:
+
+```js
+globalThis.Array === primordials.Array; // true
+
+globalThis.Array = function() {
+  return [1, 2, 3];
+};
+globalThis.Array === primordials.Array; // false
+
+primordials.Array(0); // []
+globalThis.Array(0); // [1,2,3]
+```
+
+### Prototype methods
+
+ECMAScript provides a bunch of methods available on builtin objects that are
+used to interact with JavaScript objects.
+
+```js
+const array = [1, 2, 3];
+array.push(4); // Here `push` refers to %Array.prototype.push%.
+console.log(array); // [1,2,3,4]
+
+// %Array.prototype%.push is modified in userland.
+Array.prototype.push = function push(val) {
+  return this.unshift(val);
+};
+
+array.push(5); // Now `push` refers to the modified method.
+console.log(array); // [5,1,2,3,4]
```

Comment:
Doesn't this print `[ 4, 3, 2, 1, 5 ]` when run using Node.js?

Reply:
Ahah indeed, a very good example of why we should use primordials!